File: Easy/Best_Time_to_Buy_and_Sell_Stock/Best_Time_to_Buy_and_Sell_Stock.js
class Solution {
    /**
     * @param {number[]} prices
     * @return {number}
     */
    maxProfit(prices) {
        let minBuy = prices[0];
        let maxProfit = 0;

        for (let sellPrice of prices) {
            maxProfit = Math.max(maxProfit, sellPrice - minBuy);
            minBuy = Math.min(minBuy, sellPrice);
        }

        return maxProfit;
    }
}

File: Easy/Best_Time_to_Buy_and_Sell_Stock/Best_Time_to_Buy_and_Sell_Stock.java
class Solution {
    public int maxProfit(int[] prices) {
        while (j < prices.length) {
    }
            if (prices[i] < prices[j]) {
        }
        int maxProfit = 0;
        int i = 0, j = 1;
                int profit = prices[j] - prices[i];
            } else {
                maxProfit = Math.max(maxProfit, profit);
                i = j;
            }
}
            j++;

        return maxProfit;


